app/src/processing/app/EditorLineStatus.java

```diff
@@ -110,11 +110,17 @@ public void paintComponent(Graphics graphics) {
     g.drawString(text, scale(6), baseline);
 
     g.setColor(messageForeground);
-    String tmp = I18n.format(tr("{0} on {1}"), name, serialport);
-    
-    Rectangle2D bounds = g.getFontMetrics().getStringBounds(tmp, null);
-    
-    g.drawString(tmp, size.width - (int) bounds.getWidth() - RESIZE_IMAGE_SIZE,
+
+    String statusText;
+    if (serialport != null && !serialport.isEmpty()) {
+      statusText = I18n.format(tr("{0} on {1}"), name, serialport);
+    } else {
+      statusText = name;
```

Comment:
> Also, should this check whether a port is needed for the current board?

This may be not immediately obvious, it may be that uploading with programmer may require the serial port and uploading via bootloader not, but you don't know until the user press "upload" or shift+"upload".

BTW, the purpose of this change is just to avoid dispalying `Arduino Uno on` (with the empty string) or `Arduino Uno on null` that means nothing. In this case it justs display `Arduino Uno` that is the board currently selected, I doubt that the user will ever think about the selected serial port.

Anyway if the board needs a selected port the user will be prompted on upload, if it doesn't the upload will just work. The same goes if the port is selected but not needed, the upload will work, so I think it's ok anyway.

Reply:
Fair point. One could argue that, in the case of a programmer, one should say "AVR-ISP on ttyUSB0" rather than "Arduino Uno on ttyUSB0" if the programmer is on that port, but that seems out of scope here (and not really possible to solve while the programmer and normal upload share the same port configuration).

So indeed, no change needed here.